cli/flox-activations/src/main.rs

```diff
@@ -11,18 +11,21 @@ fn main() -> Result<(), Error> {
     let args = Cli::parse();
     eprintln!("{args:?}");
 
-    let cache_dir = match args.cache_dir {
-        Some(cache_dir) => cache_dir,
+    let runtime_dir = match args.runtime_dir {
+        Some(runtime_dir) => runtime_dir,
         None => {
             let dirs = BaseDirectories::with_prefix("flox")?;
-            dirs.create_cache_directory("")?
+            match dirs.get_runtime_directory() {
+                Ok(runtime_dir) => runtime_dir.to_path_buf(),
+                Err(_) => dirs.get_cache_home(),
```

Comment:
blocking (followup okay): I think we want to join `run` to cache dir. And I think we need this logic factored out, because if we're testing or using the CLI, we'll set this in the CLI, but if we're inside a container, we'll want to determine runtime dir in `flox-activations`.

I was thinking we can't factor out all of the logic to determine cache_dir because part of that is determined by config which is a CLI only concept, is that right?

Reply:
following up with #2240